### DISCUSSION TITLE
Starting slingbox-server.exe on Windows 7 gives missing dll error

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#213] [General]
My laptop and Slingbox 500 are both on the same local network, and I wanted to run some basic tests before trying to run the server code remotely from the Slingbox. I took the unified config.ini file, edited it to have the proper info for the Slingbox 500, and tried to start run slingbox-server.exe from a command prompt - just to see if I could get a stream going. No joy... I got the following error message: "The program can't start because api-win-core-path-l1-1-0.dll is missing from your computer."

My laptop is running Windows 7, if that makes a difference.

**Comment by floormanager1** [comment]
Hi, I had a similar issue with Windows 7 and 10.  here's a thread link if that helps
https://github.com/GerryDazoo/Slinger/issues/53

**Reply by DVoorhees1957** [reply]
Thanks for the feedback. In your case, it was a 32-bit vs. 64-bit problem. I'm already running 64-bit Windows 7... I'm going to switch over to the Python version of things and see if I can make any progress.

**Comment by GerryDazoo** [comment]
A number of people have run the exe successfully on Win7. Did you google up the error?

**Reply by DVoorhees1957** [reply]
Hi Gerry, yes I did. There were only 2 or 3 results when I Googled "api-win-core-path-l1-1-0.dll" missing, and following the only link that was in English led me to a Reddit page about pirated games :-/ ... so I'm trying the Python version of things - but immediately ran into ModuleNotFoundErrors for flask and netifaces...

UPDATE: I did a pip install for flask and netifaces, and it looks like I have slingbox_server.py up and running... :-)

**Comment by DVoorhees1957** [comment]
Quick followup on this issue: First, I had confounded the problem by trying to launch slingbox_server.exe from a command prompt, instead of just right-clicking the EXE and choosing Run. As a result, when the missing DLL error occurred, my original command prompt closed - and I didn't have a chance to see the actual error (which has nothing to do with api-win-core-path-l1-1-0.dll). When I launched the EXE by right-clicking, I still get the same error message but a command prompt window (which I assume slingbox_server.exe opens) had the real error message in it:

"Error loading Python DLL 'C:\Users\Dick\AppData\Local\Temp\_MEI56002\python310.dll'. LoadLibrary: The specified module could not be found."

When I Googled "python310.dll" "missing", I found some useful results, one of which was https://stackoverflow.com/questions/72243852/pyinstaller-cant-find-a-module-error-loading-python-dll ... if you go to that link and scroll down a bit, you'll see the following:

"Ok, it was not working because I compiled the script with pyinstaller having python 3.10, but Windows 7's maximum python version is 3.8"

I don't know if the other people that have successfully run slingbox_server.exe on Windows 7 ignored the warnings on the Python download page, but I in fact paid attention to them and downloaded Python 3.8... all releases after 3.8 are supposed to not work on Windows 7 (whether or not they actually do, I can't say).

So, the straightforward thing to do would be to recompile the slingbox_server executable using Python 3.8's dll, which based on my Python 3.8 install is either python3.dll or python38.dll. Is that possible to do - just for those of us Windows 7 users that pay attention to the Python download instructions?

**Reply by sp222222** [reply]
3.08d already is backward compatible to 3.7. it should work

**Reply by DVoorhees1957** [reply]
You're confusing Gerry's Slinger version (3.08) with the Python version. This thread is specifically a conversation about the python DLL that the Slinger EXE uses (which is the Python 3.10 version of the DLL) is incompatible with Windows 7 - Windows 7 supposedly won't work with any version of Python past 3.8. So I'm asking Gerry to recompile the Slinger EXE with the Python 3.8 version of the DLL - or at least supply a Windows7-specific version of the EXE that uses the Python 3.8 DLL, not the Python 3.10 DLL...

**Reply by GerryDazoo** [reply]
I'm not going to maintain a separate exe for windows 7 but you can built one yourself. I guy over on the windows group built his own 32-bit version and you can too. Here's the command I run to "compile" the exe.

``C:\Users\Gerry\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\Scripts\pyinstaller --onefile 

Goggle up pyinstaller. Don't be intimidated by that long path name...
Although if you can build the exe you could just as easily run the python code I guess.

**Reply by DVoorhees1957** [reply]
I understand, building a separate Win7 executable is by far a more painful approach. Frankly, I was hoping that the python compiler you're using (pyinstaller?) had an option to compile for backwards compatibility with earlier Python 3 releases. (Frankly, I assume that's why there's both a python3.dll and a python38.dll in my Python 3.8 installation.)

In any event, I'm not trying to create more work for you. I am in fact using the python script directly, rather than the EXE, and will probably continue doing that. You might want to add a note to your Readme and/or release notes that there are (or might be) some Python-related issues for Windows 7 users if they use the EXE rather than the Python script...

**Comment by floormanager1** [comment]
After trying everything to get the server on an old Windows 7 laptop, I gave up and installed Windows 10.  I wish I was better at this stuff...

**Reply by DVoorhees1957** [reply]
The Python script (slingbox_server.py) seems to work just fine on my Windows 7 machine; the EXE file is what's at issue. In my case I had to install Python 3 (specifically Python v3.8) first, but Gerry's notes call that out pretty directly.
